File: backend/app/schemas/area.py
"""Area schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class AreaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: Optional[str] = None
    is_active: bool
    created_at: datetime


class AreaCreate(BaseModel):
    name: str = Field(..., max_length=100)
    description: Optional[str] = None


class AreaUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=100)
    description: Optional[str] = None
    is_active: Optional[bool] = None

File: frontend/src/types/area.ts
export interface Area {
  id: number;
  name: string;
  description: string | null;
  is_active: boolean;
  created_at: string;
}

export interface AreaCreate {
  name: string;
  description?: string;
}

export interface AreaUpdate {
  name?: string;
  description?: string;
  is_active?: boolean;
}
